cs/src/core/Bond.cs

```diff
@@ -3,7 +3,27 @@
 
 namespace Bond
 {
-    using System.Collections.Generic;
+    using System;
+    using System.Collections.Generic;
+
+    public partial class GUID
+    {
+        public static implicit operator Guid(GUID bondGuid)
+        {
+            return new Guid((int)bondGuid.Data1, (short)bondGuid.Data2, (short)bondGuid.Data3, BitConverter.GetBytes((long)bondGuid.Data4));
+        }
+
+        public static implicit operator GUID(Guid systemGuid)
+        {
+            var bytes = systemGuid.ToByteArray();
+            var bondGuid = new GUID();

```

Comment:
Minor: the following syntax avoids duplication and should work with all the compilers we use:

```
var bondGuid = new GUID
{
    Data1 = BitConveter.ToUInt32(...).
   ...
};
```